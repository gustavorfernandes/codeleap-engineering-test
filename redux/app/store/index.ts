import { configureStore } from "@reduxjs/toolkit"
import usernameReducer from "../slices/usernameSlice"

export const store = configureStore({
  reducer: {
    username: usernameReducer
  },
});

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = ReturnType<typeof store.dispatch>
