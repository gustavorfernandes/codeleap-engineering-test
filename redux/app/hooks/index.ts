import { Dispatch } from "@reduxjs/toolkit"
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux"
import { RootState } from "../store"

export const useAppDispatch = () => useDispatch<Dispatch>()
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector