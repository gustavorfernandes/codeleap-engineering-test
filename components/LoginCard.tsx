import React, { useState } from "react"
import { RootState } from "../redux/app/store"
import { useSelector, useDispatch } from "react-redux"
import { setUsername } from "../redux/app/slices/usernameSlice"

function LoginCard() {
  const [inputValue, setInputValue] = useState("")
  const userName = useSelector((state: RootState) => state.username.value)
  const dispatch = useDispatch()

  function login(e: React.FormEvent) {
    e.preventDefault()
    dispatch(setUsername(inputValue))
    clearField()
  }

  function clearField() {
    setInputValue("")
  }

  return (
    <div className="w-screen h-screen flex items-center justify-center">
      <div className="w-11/12 bg-white border border-neutral-200 p-4">
        <form
          className="flex flex-col justify-center font-roboto "
          onSubmit={(e: React.FormEvent) => {
            login(e)
          }}
        >
          <h2 className="font-bold mb-4">
            Welcome to CodeLeap network!
          </h2>
          <label
            className="text-xs"
            htmlFor="username"
          >
            Please enter your username
          </label>
          <input
            className="w-full outline-none border border-neutral-400 rounded mt-2 text-xs pl-1 py-[3px]"
            name="username"
            placeholder="John doe"
            value={inputValue}
            onChange={(e) => setInputValue(e.target.value)}
          />

          <button
            className="self-end mt-4 bg-black text-white font-bold text-xs px-5 py-1 disabled:bg-neutral-300"
            type="submit"
            disabled={!inputValue}
          >
            ENTER
          </button>
        </form>
      </div>
    </div >
  )
}

export default LoginCard
