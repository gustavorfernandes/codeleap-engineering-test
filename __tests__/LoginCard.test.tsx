import React from "react"
import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import LoginCard from "../components/LoginCard"
import { Provider } from "react-redux"
import { store } from "../redux/app/store"

describe("Login card component", () => {
  it("When the input is empty, the enter button must be disabled", () => {
    render(
      <Provider store={store} >
        <LoginCard />
      </Provider>
    )
    const input = screen.getByPlaceholderText("John doe")
    const button = screen.getByRole("button")
    expect(input).toBeInTheDocument()
    expect(button).toBeDisabled()
  })

  it("The component must render correctly", () => {
    const { container } = render(
      <Provider store={store} >
        <LoginCard />
      </Provider>
    )    
    expect(container).toMatchSnapshot()    
  })
})