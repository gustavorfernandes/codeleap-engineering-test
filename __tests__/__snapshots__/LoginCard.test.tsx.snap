// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login card component The component must render correctly 1`] = `
<div>
  <div
    class="w-screen h-screen bg-neutral-100 flex items-center justify-center"
  >
    <div
      class="w-11/12 max-w-[20rem] sm:max-w-[25rem] md:max-w-[31.25rem] bg-white border border-neutral-200 p-4 sm:p-5 md:p-6"
    >
      <form
        class="flex flex-col justify-center font-roboto"
      >
        <h2
          class="font-bold mb-4 sm:mb-5 md:mb-6 sm:text-xl md:text-[22px]"
        >
          Welcome to CodeLeap network!
        </h2>
        <label
          class="text-xs sm:text-sm md:text-base md:mb-1"
          for="username"
        >
          Please enter your username
        </label>
        <input
          class="w-full outline-none border border-neutral-400 rounded mt-2 mb-4 text-xs sm:text-[13px] md:text-sm pl-2 py-[3px]"
          id="username"
          name="username"
          placeholder="John doe"
          type="text"
          value=""
        />
        <a
          class="self-end"
          href="/dashboard"
        >
          <button
            class="bg-black hover:bg-neutral-800 transition-all text-white font-bold text-xs sm:text-sm md:text-base px-5 sm:px-7 py-1 disabled:bg-neutral-300"
            disabled=""
            type="submit"
          >
            ENTER
          </button>
        </a>
      </form>
    </div>
  </div>
</div>
`;
